app/components/dsfr/input_component.rb

```diff
@@ -0,0 +1,70 @@
+class Dsfr::InputComponent < ApplicationComponent
+  def initialize(form:, attribute:, input_type:, opts: {}, mandatory: true)
+    @form = form
+    @attribute = attribute
+    @input_type = input_type
+    @opts = opts
+    @mandatory = mandatory
+  end
+
+  def input_opts
+    @opts[:class] ||= []
+
+    @opts[:class] << 'fr-input' << 'fr-mb-0' # FIXME: after full opt-in of dsfr
+    if errors_on_attribute?
+      @opts[:class] << "fr-input--error"
+      @opts = @opts.merge(aria: { "aria-describedby" => error_message_id })
+    end
+    if @mandatory
+      @opts[:required] = true
+    end
+    @opts
+  end
+
+  def input_group_class_names
+    class_names = ['fr-input-group']
+
+    if errors_on_attribute?
+      class_names << "fr-input-group--error"
+    end
+    if errors_on_object? && !errors_on_attribute?
+      class_names << "fr-input-group--valid"
+    end
+    class_names
+  end
+
+  # tried to inline it within the template, but failed miserablt with a double render
+  def label
+    label = @form.object.class.human_attribute_name(@attribute)
+
+    if @mandatory
+      label += tag.span(" *", class: 'mandatory')
```

Comment:
```suggestion
      label += tag.span(" *", class: 'mandatory')
```

oui, je le remplace par un espace insécable :) .